packages/blockchain-wallet-v4/src/network/api/bitpay/index.js

```diff
@@ -0,0 +1,81 @@
+import axios from 'axios'
+import crypto from 'crypto'
+
+const url = 'https://bitpay.com/'
+export default () => {
+  /**
+   * Makes a request using the invoiceId and returns the raw JSON string retrieved as well as the headers
+   * @param invoiceId {string} the bitpay invoiceId
+   */
+  const getRawPaymentRequest = invoiceId =>
+    axios({
+      strictSSL: true,
+      method: 'get',
+      url: url + `i/${invoiceId}`,
+      headers: {
+        Accept: 'application/payment-request'
+      }
+    })
+      .then(response => {
+        if (response.status !== 200) {
+          throw new Error(response.data.toString())
+        }
+        return {
+          rawBody: JSON.stringify(response.data),
+          headers: response.headers
+        }
+      })
+      .catch(error => {
+        if (error) {
+          throw new Error(error)
+        }
+      })
+
+  /**
+   * Given a raw payment protocol body, parses it and validates it against the digest header
+   * @param rawPaymentRequest {object} Raw JSON object retrieved from the payment protocol server
+   */
+  const parsePaymentRequest = rawPaymentRequest => {
```

Comment:
This looks like a lot of logic to be doing inside of a request

Reply:
switched this function to a new bitpay saga